src/dotnet/ToolPackage/ToolPackageFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.Tools.Install.Tool;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class ToolPackageFactory : IToolPackageFactory
+    {
+        private const string NameOfNestedPackageDirectory = ".pkg";
```

Comment:
Nit: perhaps name this `PackageRootDirectory` or something similiar? The `NameOfNested` prefix is unnecessary imo.

Reply:
PackageRootDirectory  = ".pkg" may confused with say , c:\.pkg ?

and `name` is to tell you that is only the directory name, not the full path. And nested it indicate it will nest one more layer insterad of  tools.pkg vs tools/.pkg

Reply:
`NestedPackageDirectoryName` perhaps?  I'm very weakly opinioned on this, so if none of those suggested names speak to you, then :+1: on leaving it as is.